auth/postgres/repo.go

```diff
@@ -0,0 +1,623 @@
+		if err != nil {
+			if errRollback := tx.Rollback(); errRollback != nil {
+				err = errors.Wrap(errors.Wrap(apiutil.ErrRollbackTx, errRollback), err)
+			}
+		}
+	}()
+	res, err := tx.NamedExec(q, upm)
+	if err != nil {
+		return errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+
+	cnt, err := res.RowsAffected()
+	if err != nil {
+		return errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+	if cnt == 0 {
+		return repoerr.ErrNotFound
+	}
+
+	return nil
+}
+
+func (pr *patRepo) Remove(ctx context.Context, userID, patID string) error {
+	q := `DELETE FROM pats WHERE user_id = :user_id AND id = :id`
+	upm := dbPatPagemeta{
+		User: userID,
+		ID:   patID,
+	}
+
+	tx, err := pr.db.BeginTxx(ctx, nil)
+	if err != nil {
+		return errors.Wrap(repoerr.ErrRemoveEntity, err)
+	}
+
+	defer func() {
+		if err != nil {
+			if errRollback := tx.Rollback(); errRollback != nil {
+				err = errors.Wrap(errors.Wrap(apiutil.ErrRollbackTx, errRollback), err)
+			}
+		}
+	}()
+	res, err := tx.NamedExec(q, upm)
+	if err != nil {
+		return postgres.HandleError(repoerr.ErrRemoveEntity, err)
+	}
+
+	cnt, err := res.RowsAffected()
+	if err != nil {
+		return errors.Wrap(repoerr.ErrRemoveEntity, err)
+	}
+	if cnt == 0 {
+		return repoerr.ErrNotFound
+	}
+
+	if err := pr.cache.Remove(ctx, patID); err != nil {
+		return errors.Wrap(repoerr.ErrRemoveEntity, err)
+	}
+
+	return nil
+}
+
+func (pr *patRepo) AddScopeEntry(ctx context.Context, userID, patID string, scopes []auth.Scope) (auth.ScopesPage, error) {
+	tx, err := pr.db.BeginTxx(ctx, nil)
+	if err != nil {
+		return auth.ScopesPage{}, errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+
+	defer func() {
+		if err != nil {
+			if errRollback := tx.Rollback(); errRollback != nil {
+				err = errors.Wrap(errors.Wrap(apiutil.ErrRollbackTx, errRollback), err)
+			}
+		}
+	}()
+
+	scs := toDBScope(scopes)
+	_, err = tx.NamedQuery(saveScopeQuery, scs)
+	if err != nil {
+		return auth.ScopesPage{}, postgres.HandleError(repoerr.ErrUpdateEntity, err)
+	}
+
+	if err := tx.Commit(); err != nil {
+		return auth.ScopesPage{}, postgres.HandleError(repoerr.ErrUpdateEntity, err)
+	}
+
+	res, err := pr.retrieveScopeFromDB(ctx, patID)
+	if err != nil {
+		return auth.ScopesPage{}, err
+	}
+
+	if err := pr.cache.Save(ctx, res); err != nil {
+		return auth.ScopesPage{}, errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+
+	return auth.ScopesPage{
+		Total:  uint64(len(res)),
+		Scopes: res,
+	}, nil
+}
+
+func (pr *patRepo) RemoveScopeEntry(ctx context.Context, userID, patID string, scopes []auth.Scope) (auth.ScopesPage, error) {
+	dbscopes := toDBScope(scopes)
+
+	_, err := pr.db.NamedQueryContext(ctx, deleteScopesQuery, dbscopes)
+	if err != nil {
+		return auth.ScopesPage{}, errors.Wrap(repoerr.ErrUpdateEntity, err)
+	}
+
+	res, err := pr.retrieveScopeFromDB(ctx, patID)
+	if err != nil {
+		return auth.ScopesPage{}, err
+	}
+
+	if err := pr.cache.Save(ctx, res); err != nil {
+		return auth.ScopesPage{}, errors.Wrap(repoerr.ErrUpdateEntity, err)
```

Comment:
we need to remove scopes from cache 
```suggestion
	if err := pr.cache.Remove(ctx, res); err != nil {
```